src/rust/engine/serverset/src/lib.rs

```diff
@@ -67,18 +67,62 @@ impl<T: Clone + Send + Sync + 'static> Clone for Serverset<T> {
 #[derive(Clone, Copy, Debug)]
 #[must_use]
 pub struct HealthReportToken {
-  index: usize,
+  server_index: usize,
 }
 
-struct Inner<T> {
-  connections: Vec<Backend<T>>,
+struct Connection<T: Clone> {
+  connection: T,
+  server_index: usize,
+}
 
+struct Inner<T: Clone> {
+  // Order is immutable through the lifetime of a Serverset, but contents of the Backend may change.
```

Comment:
One thing that eases my mind is that this vec is always the same length, afaict. Would be good to mention that here.

Reply:
Done